src/game/WorldHandlers/QuestDef.h

```diff
@@ -256,6 +256,7 @@ class Quest
         bool   IsRepeatable() const { return m_SpecialFlags & QUEST_SPECIAL_FLAG_REPEATABLE; }
         bool   IsAutoComplete() const { return QuestMethod ? false : true; }
         bool   IsAllowedInRaid() const;
+        bool   HasItemsRequirement() const { return (ReqItemId[0] || ReqItemId[1] || ReqItemId[2] || ReqItemId[03]); }
```

Comment:
This function is excessive. It will be more efficient (both in code length and CPU time) to use existent function and different check,
`GetReqItemsCount() > 0`
while the `m_reqitemscount` value is calculated [here](https://github.com/mangoszero/server/blob/master/src/game/WorldHandlers/QuestDef.cpp#L151) exactly in the way needed.